std/unicode.zig

```diff
@@ -12,6 +23,47 @@ pub fn utf8ByteSequenceLength(first_byte: u8) !u3 {
     return error.Utf8InvalidStartByte;
 }
 
+/// Encodes a code point back into utf8
+/// c: the code point
+/// out: the out buffer to write to
+/// Notes: out has to have a len big enough for the bytes
+///        however this limit is dependent on the code point
+///        but giving it a minimum of 4 will ensure it will work
+///        for all code points.
+/// Errors: Will return an error if the code point is invalid.
+pub fn utf8Encode(c: u32, out: []u8) !u3 {
+    if (utf8CodepointSequenceLength(c)) |length| {
+        debug.assert(out.len >= length);
+        switch (length) {
+            1 => out[0] = u8(c), // Can just do 0 + codepoint for initial range
+            2 => {
+                // 64 to convert the codepoint into its segments
+                out[0] = u8(0b11000000 + c / 64);
```

Comment:
I think this would be a bit clearer if you used bitwise shifts and masks here instead.

Reply:
Sure, I'll update it to use shifts and masks; I had a feeling it would look worse but we'll see!

Reply:
I still like the long binary bits at the beginning of each 'out =' so I won't shorten them to hexadecimal it just makes it easier to see what it is doing, I'll remove the math for shifting/bitwise though.